tests/Microsoft.Identity.Test.Unit/ExceptionTests/MsalExceptionTests.cs

```diff
@@ -509,6 +509,65 @@ await app.AcquireTokenForClient(TestConstants.s_scope)
             }
         }
 
+        [TestMethod]
+        public async Task AuthorityInNotFoundExceptions()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                harness.HttpManager.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.AuthorityCommonTenant + TestConstants.DiscoveryEndPoint));
+                harness.HttpManager.AddMockHandlerContentNotFound(HttpMethod.Post);
+
+                ConfidentialClientApplication app = null;
+                var certificate = new X509Certificate2(
+                    ResourceHelper.GetTestResourceRelativePath("RSATestCertDotNet.pfx"));
+
+                app = ConfidentialClientApplicationBuilder
+                    .Create(TestConstants.ClientId)
+                    .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                    .WithRedirectUri(TestConstants.RedirectUri)
+                    .WithHttpManager(harness.HttpManager)
+                    .WithCertificate(certificate)
+                    .BuildConcrete();
+
+                var ex = await Assert.ThrowsExceptionAsync<MsalServiceException>(async () =>
+                {
+                    await app.AcquireTokenForClient(TestConstants.s_scope)
+                             .ExecuteAsync(CancellationToken.None)
+                             .ConfigureAwait(false);
+                }).ConfigureAwait(false);
+
+                Assert.IsTrue(ex.Message.Contains("Authority used: https://login.microsoftonline.com/common/"));
+                Assert.IsTrue(ex.Message.Contains("Token Endpoint: https://login.microsoftonline.com/common/oauth2/v2.0/token"));
+
+                //Validate that the authority is not appended with extra query parameters
+                //Validate that the region is also captured
+                Environment.SetEnvironmentVariable("REGION_NAME", TestConstants.Region);
+
+                harness.HttpManager.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.AuthorityCommonTenant + TestConstants.DiscoveryEndPoint));
+                harness.HttpManager.AddMockHandlerContentNotFound(HttpMethod.Post);

```

Comment:
can you also add a test for other error codes to see how the new method behaves?